front/src/common/Map/Settings/MapSettingsSpeedLimits.tsx

```diff
@@ -1,27 +1,44 @@
 import React, { useEffect, useState } from 'react';
-import PropTypes from 'prop-types';
 import { useSelector, useDispatch } from 'react-redux';
 import { useTranslation } from 'react-i18next';
 import { updateLayersSettings } from 'reducers/map';
 import { IoMdSpeedometer } from 'react-icons/io';
 import SwitchSNCF, { SWITCH_TYPES } from 'common/BootstrapSNCF/SwitchSNCF/SwitchSNCF';
 import SelectImprovedSNCF from 'common/BootstrapSNCF/SelectImprovedSNCF';
 import DotsLoader from 'common/DotsLoader/DotsLoader';
-import { get } from 'common/requests';
 import { setFailure } from 'reducers/main';
 import TIVsSVGFile from 'assets/pictures/layersicons/layer_tivs.svg';
 import { getInfraID } from 'reducers/osrdconf/selectors';
+import { getMap } from 'reducers/map/selectors';
+import { IconType } from 'react-icons';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
 import { FormatSwitch as SimpleFormatSwitch, Icon2SVG } from './MapSettingsLayers';
 
-function FormatSwitch(props) {
+type FormatSwitchProps = {
+  name: string;
+  icon: IconType;
+  color?: string;
+  disabled?: boolean;
+};
+
+const FormatSwitch = ({ name, icon: IconComponent, color = '', disabled }: FormatSwitchProps) => {
   const dispatch = useDispatch();
   const { t } = useTranslation(['map-settings']);
-  const { layersSettings } = useSelector((state) => state.map);
+  const { layersSettings } = useSelector(getMap);
   const infraID = useSelector(getInfraID);
-  const [speedLimitsTags, setSpeedLimitsTags] = useState(undefined);
-  const { name, icon, color, disabled = false } = props;
+  const { data: tagsList, isError: isGetSpeedLimitTagsError } =
```

Comment:
You can also access "error" to use the message for the toast below :
```suggestion
  const { data: tagsList, isError: isGetSpeedLimitTagsError, error } =
```